src/charm.py

```diff
@@ -51,6 +53,31 @@ def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
         """Config changed handler."""
         self.reconcile()
 
+    def _on_install(self, _: ops.InstallEvent) -> None:
```

Comment:
I assume the password is stored inside the container FS? If that's the case, maybe this should run on the `pebble-ready` event, as the container can restart during the charm runtime without triggering an `install` event.

Reply:
The password is installed in opensearch, so I just want to change it on clean deployment

Reply:
If the password is stored in OpenSearch, this may need to be executed after the OpenSearch integration is established, as OpenSearch is not guaranteed to be available during `install`?

Reply:
Hmm, perhaps I should defer instead of returning if the state is `None`? That's where I check the available integrations are ready